### PR TITLE
scripts: add scripts for making releases

### PR DESCRIPTION
**Pull request by perf2711**
This PR adds two scripts:
* gitRelease.ts - for a package, increments version, commits the version update to `<package>/<version>`
* release.ts - for a package, releases current version and creates a tag on HEAD or provided commit hash

Scripts have a rudimentary transaction capability - if something fails on the way, the script will try to undo all stuff it did.

sdk-core/0.0.7 was released using these scripts.

### CODE REVIEW

**Review by konraddysput — CHANGES_REQUESTED** [review]
Overall all changes look great, however, can we add a quick readme with information how to use those tools?

File: scripts/common/output.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function log(message: string) {
+    process.stderr.write(message + '\n');
```
[the remark this answers is not in the document]
Comment (perf2711) [reply scripts/common/output.ts:2]: It is by design, so you can use the output for chaining to other stuff.

File: scripts/common/packageJson.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+export async function loadPackageJson(packageJsonPath: string): Promise<PackageJson> {
+    const error = new Error(`${packageJsonPath} does not seem to be a valid package.json file`);
+
+    let packageJson: Partial<PackageJson>;
+    try {
+        packageJson = JSON.parse(await fs.promises.readFile(packageJsonPath, 'utf-8')) as Partial<PackageJson>;
+    } catch {
+        throw error;
```
[the remark this answers is not in the document]
Comment (perf2711) [reply scripts/common/packageJson.ts]: Done.

File: scripts/gitRelease.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    const options = ['--dry-run', '--no-push', '--no-commit', '--no-checkout', '--no-add'] as const;
+
+    const argv = process.argv.slice(2);
+    const [packageJsonPath, versionOrRelease, identifier] = argv.filter((v) => !options.includes(v as never));
+    const optionValues = options.reduce((val, k) => {
+        val[k] = argv.includes(k);
+        return val;
+    }, {} as Record<(typeof options)[number], boolean>);
+
+    if (!packageJsonPath) {
```
[the remark this answers is not in the document]
Comment (perf2711) [reply scripts/gitRelease.ts:68]: Done.

**Review by konraddysput — APPROVED** [review]

File: scripts/common/output.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (konraddysput) [reply scripts/common/output.ts:2]: yeah but it has to be stderr? You cannot use stdout?

**Pull request merged**